example/figure_eight_plan_6dof.py

```diff
@@ -0,0 +1,172 @@
+# Python standard lib
+import os
+import sys
+import pathlib
+
+# ROS
+# import rospy
+# from sensor_msgs.msg import JointState
+
+# PyBullet
+import pybullet_api
+
+# OpTaS
+import optas
+
+
+class Planner:
+
+
+    def __init__(self):
+
+        cwd = pathlib.Path(__file__).parent.resolve() # path to current working directory
+        pi = optas.np.pi  # 3.141...
+        self.T = 50 # no. time steps in trajectory
+        link_ee = 'end_effector_ball'  # end-effector link name
+        self.Tmax = 10.  # trajectory of 5 secs
+        t = optas.linspace(0, self.Tmax, self.T)
+        self.dt = float((t[1] - t[0]).toarray()[0, 0])  # time step
+
+        # Setup robot
+        urdf_filename = os.path.join(cwd, 'robots', 'kuka_lwr', 'kuka_lwr.urdf')
+        self.kuka = optas.RobotModel(
+            urdf_filename=urdf_filename,
+            time_derivs=[0, 1],  # i.e. joint position/velocity trajectory
+            param_joints=['lwr_arm_0_joint'],
+        )
+        self.kuka_name = self.kuka.get_name()
+
+        # Setup optimization builder
+        builder = optas.OptimizationBuilder(T=self.T, robots=[self.kuka])
+
+        # Setup parameters
+        qc = builder.add_parameter('qc', self.kuka.ndof)  # current robot joint configuration
+
+        # Constraint: initial configuration
+        builder.initial_configuration(self.kuka_name, qc[self.kuka.optimized_joint_indexes])
```

Comment:
This is an example of usage. It just happened that the current configuration in this example was zero. But had it been something else it would be necessary to specify the initial configuration which might change for each call of the optimizer (after build).

Reply:
Yes, but what you are specifying here does not impact the decision variables. You are making a constraint
```
g(p) = 0
```
rather than
```
g(x; p) = 0
```
If you want to put a non-zero configuration then this will not happen - this is because you are specifying the constraint incorrectly.